controllers/ytsaurus_controller_test.go

```diff
@@ -400,6 +459,66 @@ var _ = Describe("Basic test for Ytsaurus controller", func() {
 			By("Check that access control object 'nobody' in namespace 'queries' exists")
 			Expect(ytClient.NodeExists(ctx, ypath.Path("//sys/access_control_object_namespaces/queries/nobody"), nil)).Should(Equal(true))
 		})
+
+		It("Should create ytsaurus with remote exec nodes and execute a job", func() {
+			By("Creating a Ytsaurus resource")
+			ctx := context.Background()
+
+			namespace := "remoteexec"
+
+			ytsaurus := ytv1.CreateMinimalYtsaurusResource(namespace)
+			// Ensure that no local exec nodes exist, only remote ones (which will be created later).
+			ytsaurus.Spec.ExecNodes = []ytv1.ExecNodesSpec{}
+			ytsaurus.Spec.DataNodes = []ytv1.DataNodesSpec{{
+				InstanceSpec: ytv1.CreateDataNodeInstanceSpec(1),
+			}}
+			g := ytconfig.NewGenerator(ytsaurus, "local")
+
+			remoteYtsaurus := &ytv1.RemoteYtsaurus{
+				ObjectMeta: v1.ObjectMeta{
+					Name:      ytv1.RemoteResourceName,
+					Namespace: namespace,
+				},
+				Spec: ytv1.RemoteYtsaurusSpec{
+					MasterConnectionSpec: ytv1.MasterConnectionSpec{
+						CellTag: ytsaurus.Spec.PrimaryMasters.CellTag,
+						HostAddresses: []string{
+							"ms-0.masters.remoteexec.svc.cluster.local",
+						},
+					},
+				},
+			}
+
+			remoteNodes := &ytv1.RemoteExecNodes{
+				ObjectMeta: v1.ObjectMeta{Name: ytv1.RemoteResourceName, Namespace: namespace},
+				Spec: ytv1.RemoteExecNodesSpec{
+					RemoteClusterSpec: &corev1.LocalObjectReference{
+						Name: ytv1.RemoteResourceName,
+					},
+					CommonSpec: ytv1.CommonSpec{
+						CoreImage: ytv1.CoreImageFirst,
+					},
+					ExecNodesSpec: ytv1.ExecNodesSpec{
+						InstanceSpec: ytv1.CreateExecNodeInstanceSpec(),
+					},
+				},
+			}
+
+			defer deleteYtsaurus(ctx, ytsaurus)
+			runYtsaurus(ytsaurus)
+
+			defer deleteRemoteYtsaurus(ctx, remoteYtsaurus)
+			runRemoteYtsaurus(remoteYtsaurus)
+
+			defer deleteRemoteExecNodes(ctx, remoteNodes)
+			runRemoteExecNodes(remoteNodes)
+
+			By("Creating ytsaurus client")
+			ytClient := getYtClient(g, namespace)
+
+			By("Running sort operation to ensure exec node works")
+			runAndCheckSortOperation(ytClient)
```

Comment:
Let's return operation from this call, and make sure that all jobs within operation were indeed run on remote nodes? This can be done by list_jobs(op.id) call, and maybe get_job(job.id) for each job.

Reply:
For some reason `ytClient.ListJobs` returns empty list of jobs. Any ideas why that could happen? For now i've tried to pass `JobState: &yt.JobCompleted` but result is the same.

Reply:
Were there any jobs at all? Can you see them in scheduler logs?

Reply:
Well, my table was sorted as far as i can see.
I'll check the logs.

Reply:
In order to debug something like that, it is convenient to insert sleep(10000) after operation completion and connect to cluster using yt CLI, try that also.